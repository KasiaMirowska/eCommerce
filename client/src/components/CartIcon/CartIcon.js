import React from 'react';
import './CartIcon.styles.scss';
// import {
//     CartContainer,
//     ShoppingIcon,
//     ItemCountContainer
// } from './CartIcon.styles';
import { ReactComponent as ShoppingIcon } from '../../assets/cart.svg';
import { connect } from 'react-redux';
import { toggleCartHidden } from '../../redux/cart/cart.actions'
import { selectCartItemsCount } from '../../redux/cart/cart.selectors';

const mapDispatchToProps = dispatch => ({
    toggleCartHidden: () => dispatch(toggleCartHidden())
})

const mapStateToProps = state => {
    return ({
        //when pulling just a part of state (here just quantity), we're creating a selector
        itemCount: selectCartItemsCount(state)
    })
}

export const CartIcon = ({ toggleCartHidden, itemCount }) => {

    return (
        <div className='cart-icon' id='cart-Icon' onClick={toggleCartHidden}>
            <ShoppingIcon className='shopping-icon' />
            <span className='item-count'>{itemCount}</span>
        </div>
        // <CartContainer onClick={toggleCartHidden}>
        //     <ShoppingIcon />
        //     <ItemCountContainer>{itemCount}</ItemCountContainer>
        // </CartContainer>
    )

}

export default connect(mapStateToProps, mapDispatchToProps)(CartIcon);