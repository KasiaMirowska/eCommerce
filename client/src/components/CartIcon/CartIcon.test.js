import React from 'react';
import { shallow, mount } from 'enzyme';
import { CartIcon } from '../../components/CartIcon/CartIcon';


describe('CartIcon component', () => {

    let wrapper;
    let mockToggleCartHidden;
    beforeEach(() => {
        mockToggleCartHidden = jest.fn();

        const props = {
            itemCount: 3,
            toggleCartHidden: mockToggleCartHidden
        };

        wrapper = shallow(<CartIcon {...props} />);
    });
    it('renders as expected', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('onClick calls toggleCartHidden', () => {
        // wrapper.find('CartContainer').simulate('click');
        wrapper.find('.cart-icon').simulate('click')
        expect(mockToggleCartHidden).toHaveBeenCalled();
    })

    it('should render itemCount as text', () => {
        let amount = Number( wrapper.find('.item-count').text() );
        expect(amount).toBe(3)
    })
})