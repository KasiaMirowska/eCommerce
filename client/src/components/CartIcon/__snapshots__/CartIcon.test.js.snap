// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartIcon component renders as expected 1`] = `ShallowWrapper {}`;
